src/libs/EmojiUtils.js

```diff
@@ -259,6 +259,42 @@ const getEmojiCodeWithSkinColor = (item, preferredSkinToneIndex) => {
     return code;
 };
 
+/**
+ * Extracts emojis from a given text.
+ *
+ * @param {String} text - The text to extract emojis from.
+ * @returns {Object[]} An array of emoji codes.
+ */
+function extractEmojis(text) {
+    const emojis = [];
+
+    if (!text) {
+        return emojis;
+    }
+
+    let parseEmojis = text.match(CONST.REGEX.EMOJIS);
+
+    if (!parseEmojis) {
+        return emojis;
+    }
+
+    parseEmojis = [...new Set(parseEmojis)]; // Some emojis are repeated with different skin tones
+
+    const alreadyParsedEmojis = new Set();
+
+    for (let i = 0; i < parseEmojis.length; i++) {
```

Comment:
This could be cleaned up a bit IMO...

We are making 2 sets when we could get by with just one...

Does something like this work?

```js
parseEmojis = [...new Set(parseEmojis)];
return _.reduce(parseEmojis, (finalEmojis, parsedEmoji) => {
    const emoji = Emojis.emojiCodeTableWithSkinTones[parsedEmoji];
    return emoji ? [...finalEmojis, emoji] : finalEmojis;
}, []);
```

Reply:
@marcaaron Issue is that the `emoji` could be repeated as i can have same emoji with different skin tones in `parseEmojis`, as first set clears the duplicates and second check's if this emoji was already parsed or not.

Reply:
Can you explain it another way? I'm not understanding. @parasharrajat can the code or documentation be improved a bit here somehow?